### PR TITLE
Allow assigning issues to devices

### PR DESCRIPTION
**Pull request by hughsie**
This allows us to show in the tools if a device is currently affected
by a specific CVE. For instance, we could inform the user that a device
requires a critical firmware update that is being actively exploited.

Note, this also means we can show the user a firmware update is now
required, even though the firmware may not be available on the LVFS.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [x] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by superm1** [comment]
> Note, this also means we can show the user a firmware update is now
required, even though the firmware may not be available on the LVFS.

I don't really understand the mechanics of this.  How can you know that there is a CVE to be fixed without metadata telling you there is one?

On another note; depending upon the longer roadmap for this feature it might make sense to have a dedicated dbus interface for getting this information.  "Show me all the CVEs currently affected by the firmware on this machine".  It would mean that software other than traditional fwupd clients might consume it.

**Review by superm1 — APPROVED** [review]

**Comment by hughsie** [comment]
>  How can you know that there is a CVE to be fixed without metadata telling you there is one

Adding the metadata to the quirk files, e.g.

	[EK0800JVYPN_____________________________HPD5]
	Issue = MTX-6f54e8621f4c490a918defdab6
	[EK0800JVYPN_____________________________HPD6]
	Issue = MTX-6f54e8621f4c490a918defdab6
	[EO1600JVYPP_____________________________HPD5]
	Issue = MTX-6f54e8621f4c490a918defdab6
	[EO1600JVYPP_____________________________HPD6]
	Issue = MTX-6f54e8621f4c490a918defdab6
	[MK0800JVYPQ_____________________________HPD5]
	Issue = MTX-6f54e8621f4c490a918defdab6
	[MK0800JVYPQ_____________________________HPD6]
	Issue = MTX-6f54e8621f4c490a918defdab6
	[MO1600JVYPR_____________________________HPD5]
	Issue = MTX-6f54e8621f4c490a918defdab6
	[MO1600JVYPR_____________________________HPD6]
	Issue = MTX-6f54e8621f4c490a918defdab6

**Comment by superm1** [comment]
> > How can you know that there is a CVE to be fixed without metadata telling you there is one
> 
> Adding the metadata to the quirk files, e.g.
> 
> ```
> [EK0800JVYPN_____________________________HPD5]
> Issue = MTX-6f54e8621f4c490a918defdab6
> [EK0800JVYPN_____________________________HPD6]
> Issue = MTX-6f54e8621f4c490a918defdab6
> [EO1600JVYPP_____________________________HPD5]
> Issue = MTX-6f54e8621f4c490a918defdab6
> [EO1600JVYPP_____________________________HPD6]
> Issue = MTX-6f54e8621f4c490a918defdab6
> [MK0800JVYPQ_____________________________HPD5]
> Issue = MTX-6f54e8621f4c490a918defdab6
> [MK0800JVYPQ_____________________________HPD6]
> Issue = MTX-6f54e8621f4c490a918defdab6
> [MO1600JVYPR_____________________________HPD5]
> Issue = MTX-6f54e8621f4c490a918defdab6
> [MO1600JVYPR_____________________________HPD6]
> Issue = MTX-6f54e8621f4c490a918defdab6
> ```

Oh so it won't flow from LVFS - got it.

**Comment by hughsie** [comment]
> Yeah that makes sense

I've added that; how does that look?

File: src/fu-util-common.c [hunk 1]
```diff
@@ -1501,6 +1502,14 @@ fu_util_device_to_string(FwupdDevice *dev, guint idt)
 			fu_common_string_append_kv(str, idt + 1, "", bullet);
 		}
 	}
+	for (guint i = 0; i < issues->len; i++) {
+		const gchar *issue = g_ptr_array_index(issues, i);
+		fu_common_string_append_kv(str,
+					   idt + 1,
+					   /* TRANSLATORS: issue fixed with the release, e.g. CVE */
+					   i == 0 ? ngettext("Issue", "Issues", issues->len) : "",
```
[the remark this answers is not in the document]
Comment (hughsie) [reply src/fu-util-common.c]: `Security Issue` works better, indeed. Fixed.

[the remark this answers is not in the document]
Comment (hughsie) [reply src/fu-util-common.c]: Actually, maybe it's not always security. What if it was a data loss issue? That's not technically a security issue.

[the remark this answers is not in the document]
Comment (superm1) [reply src/fu-util-common.c]: What example have you seen that would fit that from firmware? Like device stability? Would an OEM actually publish something that is not a security advisory for that?

[the remark this answers is not in the document]
Comment (hughsie) [reply src/fu-util-common.c]: e.g. https://support.hpe.com/hpesc/public/swd/detail?swItemId=MTX-6f54e8621f4c490a918defdab6#tab4

where all the disks in a raid array fail almost simultaneously...

**Pull request merged**